File: Data Structures & Algorithms/jump-game/submission-3.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        currGoal = len(nums)-1

        for i in range(len(nums) - 2, -1, -1):
            if i + nums[i] >= currGoal:
                currGoal = i
        
        return currGoal == 0

            

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> counts;
        vector<vector<int>> freqs(nums.size() + 1);
        vector<int> result;

        for (int num : nums)
        {
            counts[num]++;
        }

        for (auto [key, val] : counts)
        {
            freqs[val].push_back(key);
        }

        for (int i = freqs.size() - 1; i > 0; i--)
        {
            for (int val : freqs[i])
            {
                result.push_back(val);

                if (result.size() == k)
                {
                    return result;
                }
            }
        }

        return result;
    }
};
